src/ui/public/agg_response/point_series/_tooltip_formatter.js

```diff
@@ -18,45 +18,49 @@
  */
 
 import $ from 'jquery';
+import { getFormat } from '../../visualize/loader/pipeline_helpers/utilities';
 
 export function PointSeriesTooltipFormatter($compile, $rootScope) {
 
   const $tooltipScope = $rootScope.$new();
   const $tooltip = $(require('ui/agg_response/point_series/_tooltip.html'));
   $compile($tooltip)($tooltipScope);
 
-  return function tooltipFormatter(event) {
-    const data = event.data;
-    const datum = event.datum;
-    if (!datum) return '';
-
-    const details = $tooltipScope.details = [];
-
-    const addDetail = (label, value) => details.push({ label, value });
-
-    datum.extraMetrics.forEach(metric => {
-      addDetail(metric.label, metric.value);
-    });
-
-    if (datum.x) {
-      addDetail(data.xAxisLabel, data.xAxisFormatter(datum.x));
-    }
-    if (datum.y) {
-      const value = datum.yScale ? datum.yScale * datum.y : datum.y;
-      addDetail(data.yAxisLabel, data.yAxisFormatter(value));
-    }
-    if (datum.z) {
-      addDetail(data.zAxisLabel, data.zAxisFormatter(datum.z));
-    }
-    if (datum.series && datum.parent) {
-      const dimension = datum.parent;
-      addDetail(dimension.title, dimension.fieldFormatter(datum.series));
-    }
-    if (datum.tableRaw) {
-      addDetail(datum.tableRaw.title, datum.tableRaw.value);
-    }
-
-    $tooltipScope.$apply();
-    return $tooltip[0].outerHTML;
+  return function () {
```

Comment:
Why is the outer function necessary here?

Reply:
we need same format as in hierarchical data tooltip formatter